pkg/eth/signatures.go

```diff
@@ -21,22 +21,27 @@ func Sign(privKey *ecdsa.PrivateKey, hash util.Hash) (chain.Signature, error) {
 }
 
 func ValidateSignature(hash, signature []byte, address common.Address) error {
+        // 2 Us
 	ethHash := util.GethHash(hash)
 
+        // 200 ns
 	sigCopy := make([]byte, len(signature))
 	copy(sigCopy, signature)
 	if len(sigCopy) == 65 && sigCopy[64] > 26 {
 		sigCopy[64] -= 27
 	}
 
+        // 220 Us
 	pubKey, err := crypto.SigToPub(ethHash, sigCopy)
```

Comment:
Basically 70% of the work happens right here on this line.

Reply:
If you do the rough math here: If this takes say on average 200 us. That means about 5 transactions processed a millisecond. And in a 100ms block with 16-cores (my setup0 parallelizing the processing you get: 5*100*16 = 8000 TPS. I'm basically this figure plus/minus overhead.